tests/Microsoft.Identity.Test.Unit/PublicApiTests/ConfidentialClientApplicationTests.cs

```diff
@@ -1080,6 +1080,55 @@ public void EnsurePublicApiSurfaceExistsOnInterface()
             PublicClientApplicationTests.CheckBuilderCommonMethods(byRefreshTokenBuilder);
         }
 
+        [TestMethod]
+        public async Task ConfidentialClientSuggestedExpiryAsync()
+        {
+            using (var httpManager = new MockHttpManager())
+            {
+                httpManager.AddInstanceDiscoveryMockHandler();
+
+                var app = ConfidentialClientApplicationBuilder.Create(TestConstants.ClientId)
+                                                              .WithAuthority(new Uri(ClientApplicationBase.DefaultAuthority), true)
+                                                              .WithRedirectUri(TestConstants.RedirectUri)
+                                                              .WithClientSecret(TestConstants.ClientSecret)
+                                                              .WithHttpManager(httpManager)
+                                                              .BuildConcrete();
+
+                long expectedTimeDiff = 1000;
+                httpManager.CreateSuccessfulClientCredentialTokenResponseWithExpiry(expectedTimeDiff.ToString());
+
+                var appCacheAccess = app.AppTokenCache.RecordAccess();
+                var userCacheAccess = app.UserTokenCache.RecordAccess();
+
+                app.AppTokenCache.SetAfterAccess((args) => 
+                {
+                    if (args.SuggestedCacheExpiry != null)
```

Comment:
`if` could be replaced with an Assert that `SuggestedCacheExpiry` is not null (when HasStateChanged true).